examples/decoding/plot_decoding_time_generalization.py

```diff
@@ -12,80 +12,81 @@
 The idea is to learn at one time instant and assess if the decoder
 can predict accurately over time.
 """
-# Authors: Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+# Authors: Jean-Remi King <jeanremi.king@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#          Denis Engemann <denis.engemann@gmail.com>
 #
 # License: BSD (3-clause)
 
 import numpy as np
-import matplotlib.pyplot as plt
 
 import mne
-from mne.datasets import spm_face
-from mne.decoding import time_generalization
+from mne.datasets import sample
+from mne.decoding import GeneralizationAcrossTime
 
 print(__doc__)
 
-data_path = spm_face.data_path()
-
 ###############################################################################
-# Load and filter data, set up epochs
+# Preprocess data
 
-raw_fname = data_path + '/MEG/spm/SPM_CTF_MEG_example_faces%d_3D_raw.fif'
+data_path = sample.data_path()
+# Load and filter data, set up epochs
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+events_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw-eve.fif'
+raw = mne.io.Raw(raw_fname, preload=True)
+picks = mne.pick_types(raw.info, meg=True, exclude='bads')  # Pick MEG channels
+raw.filter(1, 30, method='fft')  # Band pass filtering signals
+events = mne.read_events(events_fname)
+event_id = {'AudL': 1, 'AudR': 2, 'VisL': 3, 'VisR': 4}
+decim = 2  # decimate to make the example faster to run
+epochs = mne.Epochs(raw, events, event_id, -0.050, 0.400, proj=True,
+                    picks=picks, baseline=None, preload=True,
+                    reject=dict(mag=5e-12), decim=decim, verbose=False)
 
-raw = mne.io.Raw(raw_fname % 1, preload=True)  # Take first run
-raw.append(mne.io.Raw(raw_fname % 2, preload=True))  # Take second run too
+###############################################################################
+# Generalization across time (GAT)
 
-picks = mne.pick_types(raw.info, meg=True, exclude='bads')
-raw.filter(1, 45, method='iir')
+# The function implements the method used in:
+# King, Gramfort, Schurger, Naccache & Dehaene, "Two distinct dynamic modes
+# subtend the detection of unexpected sounds", PLOS ONE, 2013
 
-events = mne.find_events(raw, stim_channel='UPPT001')
-event_id = {"faces": 1, "scrambled": 2}
-tmin, tmax = -0.1, 0.5
+# Define events of interest
+events = epochs.events[:, 2]
```

Comment:
a variable with 1 column should not be called events which is expected to have 3